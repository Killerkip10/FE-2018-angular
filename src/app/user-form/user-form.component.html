<mat-card>
  <mat-toolbar>{{'USER-PROFILE.FORM.HEADER' | translate}}</mat-toolbar>
  <mat-card-content>
    <form [formGroup]="userForm" (ngSubmit)="onSubmit($event)">
      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="login"
            [placeholder]="'USER-PROFILE.FORM.LOGIN' | translate">
          <mat-spinner *ngIf="login.pending" matSuffix [diameter]="18"></mat-spinner>
        </mat-form-field>
        <p *ngIf="login.errors?.login">{{login.errors?.login.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="password"
            [placeholder]="'USER-PROFILE.FORM.PASSWORD' | translate">
        </mat-form-field>
        <p *ngIf="password.errors?.password">{{password.errors?.password.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="name"
            [placeholder]="'USER-PROFILE.FORM.NAME' | translate">
          <mat-spinner *ngIf="name.pending" matSuffix [diameter]="18"></mat-spinner>
        </mat-form-field>
        <p *ngIf="name.errors?.name">{{name.errors?.name.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="age"
            [placeholder]="'USER-PROFILE.FORM.AGE' | translate">
        </mat-form-field>
        <p *ngIf="age.errors?.age">{{age.errors?.age.message | translate:{from: 18, to: 65} }}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="dateOfBirth"
            [placeholder]="'USER-PROFILE.FORM.DATE-OF-BIRTH' | translate">
        </mat-form-field>
        <p *ngIf="dateOfBirth.errors?.date">{{dateOfBirth.errors?.date.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="dateOfFirstLogin"
            [placeholder]="'USER-PROFILE.FORM.DATE-OF-FIRST-LOGIN' | translate">
        </mat-form-field>
        <p *ngIf="dateOfFirstLogin.errors?.date">{{dateOfFirstLogin.errors?.date.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <input
            matInput
            type="text"
            formControlName="dateOfNextNot"
            [placeholder]="'USER-PROFILE.FORM.DATE-OF-NEXT-NOT' | translate">
        </mat-form-field>
        <p *ngIf="dateOfNextNot.errors?.date">{{dateOfNextNot.errors?.date.message | translate}}</p>
      </div>

      <div>
        <mat-form-field>
          <textarea
            matInput
            formControlName="information"
            [placeholder]="'USER-PROFILE.FORM.INFORMATION' | translate">
          </textarea>
        </mat-form-field>
      </div>

      <div class="legend">
        <div>{{'USER-PROFILE.FORM.DATE-FORMATS' | translate}}: </div>
        <div>YYYY/MM/DD | DD MMMM YYYY | DD-MMM-YY</div>
      </div>

      <button mat-flat-button color="primary" type="submit" [disabled]="!userForm.valid">{{'SEND' | translate}}</button>
    </form>
  </mat-card-content>
</mat-card>
