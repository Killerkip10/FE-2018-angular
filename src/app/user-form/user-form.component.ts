import {Component, Input, Output, EventEmitter, OnInit, OnChanges, SimpleChanges} from '@angular/core';
import {Validators, FormBuilder} from '@angular/forms';
import * as moment from 'moment';

import {User} from '../models';
import {
  NameAsyncValidatorService,
  LoginAsyncValidatorService,
  HttpValidatorService,
  ageValidator,
  dateValidator,
  passwordValidator
} from './validators';

interface Options {
  nameForm: string;
  send: string;
  reset: boolean;
  login: boolean;
  dateOfFirstLogin: boolean;
}

const initialUser = {
  login: '',
  password: '',
  name: '',
  age: '',
  dateOfBirth: '',
  dateOfFirstLogin: '',
  dateOfNextNot: '',
  information: ''
};

@Component({
  selector: 'app-user-form',
  templateUrl: './user-form.component.html',
  styleUrls: ['./user-form.component.scss'],
  providers: [
    NameAsyncValidatorService,
    LoginAsyncValidatorService,
    HttpValidatorService,
  ]
})
export class UserFormComponent implements OnInit, OnChanges {
  @Input() options: Options = {} as Options;
  @Input() user: User = initialUser as User;
  @Output() submit = new EventEmitter<User>();

  public userForm;

  constructor(
    private nameAsyncValidService: NameAsyncValidatorService,
    private loginAsyncValidService: LoginAsyncValidatorService,
    private fb: FormBuilder
  ) {}

  ngOnInit(): void {
    this.userForm = this.fb.group({
      login: [
        {value: this.user.login, disabled: this.options.login},
        [Validators.required],
        [this.loginAsyncValidService.loginAsyncValidator()]
      ],
      password: [
        this.user.password,
        [Validators.required, passwordValidator]
      ],
      name: [
        this.user.name,
        [Validators.required],
        [this.nameAsyncValidService.nameAsyncValidator()]
      ],
      age: [
        this.user.age,
        [Validators.required, ageValidator]
      ],
      dateOfBirth: [
        this.dateFormat(this.user.dateOfBirth),
        [Validators.required, dateValidator]
      ],
      dateOfFirstLogin: [
        {value: this.dateFormat(this.user.dateOfFirstLogin), disabled: this.options.dateOfFirstLogin},
        [Validators.required, dateValidator]
      ],
      dateOfNextNot: [
        this.dateFormat(this.user.dateOfNextNot),
        [Validators.required, dateValidator]
      ],
      information: [this.user.information]
    });

    this.nameAsyncValidService.initialName = this.user.name;
    this.loginAsyncValidService.initialLogin = this.user.login;
  }
  ngOnChanges(changes: SimpleChanges): void {
    if (!changes.user || changes.user.firstChange) {
      return;
    }

    this.nameAsyncValidService.initialName = this.user.name;
    this.loginAsyncValidService.initialLogin = this.user.login;
    this.userForm.patchValue({
      ...this.user,
      dateOfBirth: this.dateFormat(this.user.dateOfBirth),
      dateOfNextNot: this.dateFormat(this.user.dateOfNextNot),
      dateOfFirstLogin: this.dateFormat(this.user.dateOfFirstLogin)
    });
  }

  public get login(){
    return this.userForm.get('login');
  }
  public get password(){
    return this.userForm.get('password');
  }
  public get name(){
    return this.userForm.get('name');
  }
  public get age(){
    return this.userForm.get('age');
  }
  public get dateOfBirth(){
    return this.userForm.get('dateOfBirth');
  }
  public get dateOfFirstLogin(){
    return this.userForm.get('dateOfFirstLogin');
  }
  public get dateOfNextNot(){
    return this.userForm.get('dateOfNextNot');
  }
  public get information(){
    return this.userForm.get('information');
  }

  public onSubmit(event): void {
    this.submit.emit({...this.user, ...this.userForm.value});
    event.stopPropagation();

    if (this.options.reset) {
      this.userForm.reset(initialUser);
    }
  }

  private dateFormat(date: string, format: string = 'YYYY/MM/DD'): string {
    return date ? moment(date).format(format) : '';
  }
}
