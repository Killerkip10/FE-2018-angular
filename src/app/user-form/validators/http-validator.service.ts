import {Injectable} from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {catchError, map} from 'rxjs/operators';
import {of} from 'rxjs';

import {urlConfig} from '../../config';

const options = {
  headers: new HttpHeaders({'content-type': 'application/json'}),
  observe: 'response',
  withCredentials: true,
  params: null
};

@Injectable()
export class HttpValidatorService {
  constructor(private http: HttpClient) {}

  public checkName(name: string) {
    return this.http.get(urlConfig.checkName + name, options as object)
      .pipe(
        map(() => null),
        catchError(() => of({name: {message: 'ERROR.EXIST-NAME'}}))
      );
  }
  public checkLogin(login: string) {
    return this.http.get(urlConfig.checkLogin + login, options as object)
      .pipe(
        map(() => null),
        catchError(() => of({login: {message: 'ERROR.EXIST-LOGIN'}}))
      );
  }
}
