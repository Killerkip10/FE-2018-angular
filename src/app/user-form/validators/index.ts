export * from './name.async-validator.service';
export * from './age.validator';
export * from './date.validation';
export * from './password.validator';
export * from './login.async-validator.service';
export * from './http-validator.service';
