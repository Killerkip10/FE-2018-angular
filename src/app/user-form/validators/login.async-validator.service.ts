import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {AbstractControl, ValidationErrors, ValidatorFn} from '@angular/forms';
import {Observable, of} from 'rxjs';
import {catchError, delay, map, mergeMap} from 'rxjs/operators';

import {HttpValidatorService} from './http-validator.service';

@Injectable()
export class LoginAsyncValidatorService {
  public initialLogin: string;

  constructor (
    private http: HttpClient,
    private httpValidService: HttpValidatorService
  ) {}

  public loginAsyncValidator(): ValidatorFn {
    return (control: AbstractControl): Observable<ValidationErrors | null> =>
      of(null)
        .pipe(
          delay(1000),
          map(() => this.checkLatinFigurSymbols(control.value)),
          mergeMap(() => this.checkLoginHttp(control.value)),
          catchError(err => of({login: {message: err.message}}))
        );
  }

  private checkLatinFigurSymbols(word: string): null {
    for (let i = 0; i < word.length; i++) {
      const symbol = word.charCodeAt(i);

      if (!(symbol >= 65 && symbol <= 90) && !(symbol >= 97 && symbol <= 122) && !(symbol >= 48 && symbol <= 57)) {
        throw new Error('ERROR.LATIN-FIGURE');
      }
    }

    return null;
  }
  private checkLoginHttp(login: string) {
    if (login === this.initialLogin) {
      return of(null);
    } else {
      return this.httpValidService.checkLogin(login);
    }
  }
}
