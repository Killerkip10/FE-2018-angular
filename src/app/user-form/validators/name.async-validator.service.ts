import {Injectable} from '@angular/core';
import {ValidationErrors, AbstractControl, ValidatorFn} from '@angular/forms';
import {Observable} from 'rxjs';
import {catchError, delay, map, mergeMap} from 'rxjs/internal/operators';
import {of} from 'rxjs';

import {HttpValidatorService} from './http-validator.service';

@Injectable()
export class NameAsyncValidatorService {
  public initialName: string;

  constructor(public httpValidService: HttpValidatorService) {}

  public nameAsyncValidator(): ValidatorFn {
    return (control: AbstractControl): Observable<ValidationErrors | null> =>
      of(null)
        .pipe(
          delay(1000),
          map(() => this.checkLength(control.value, 2)),
          map(() => this.checkCamelSensitive(control.value)),
          map(() => this.checkLatinLetter(control.value)),
          mergeMap(() => this.checkNameHttp(control.value)),
          catchError(err => of({name: {message: err.message}}))
        );
  }

  private findWords(value: string): string[] {
    return value.trim().split(' ').filter(Boolean);
  }
  private checkLength(name: string, maxLength: number): null {
    if (this.findWords(name).length > maxLength) {
      throw new Error('ERROR.WORDS-COUNT');
    } else {
      return null;
    }
  }
  private checkCamelSensitive(name: string): null {
    for (const word of this.findWords(name)){
      if (word[0].toUpperCase() !== word[0]) {
        throw new Error('ERROR.CAMEL');
      }
    }

    return null;
  }
  private checkLatinLetter(name: string): null {
    const str = this.findWords(name).join('');

    for (let i = 0; i < str.length; i++) {
      const symbol = str.charCodeAt(i);

      if (!(symbol >= 65 && symbol <= 90) && !(symbol >= 97 && symbol <= 122)) {
        throw new Error('ERROR.LATIN');
      }
    }

    return null;
  }
  private checkNameHttp(name: string) {
    if (name === this.initialName) {
      return of(null);
    } else {
      return this.httpValidService.checkName(name);
    }
  }
}

