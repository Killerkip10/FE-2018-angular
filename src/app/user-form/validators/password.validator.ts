import {ValidationErrors, AbstractControl} from '@angular/forms';

export function passwordValidator(control: AbstractControl): ValidationErrors | null {
  const password = control.value;

  for (let i = 0; i < password.length; i++) {
    const symbol = password.charCodeAt(i);

    if (!(symbol >= 65 && symbol <= 90) && !(symbol >= 97 && symbol <= 122) && !(symbol >= 48 && symbol <= 57)) {
      return {password: {message: 'ERROR.LATIN-FIGURE'}};
    }
  }

  return null;
}
