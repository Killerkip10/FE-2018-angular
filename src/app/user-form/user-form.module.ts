import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {TranslateModule} from '@ngx-translate/core';
import {ReactiveFormsModule} from '@angular/forms';

import {UserFormComponent} from './user-form.component';
import {
  MatFormFieldModule,
  MatInputModule,
  MatCardModule,
  MatButtonModule,
  MatToolbarModule,
  MatProgressSpinnerModule
} from '@angular/material';

@NgModule({
  imports: [
    CommonModule,
    TranslateModule,
    ReactiveFormsModule,

    MatFormFieldModule,
    MatInputModule,
    MatCardModule,
    MatButtonModule,
    MatToolbarModule,
    MatProgressSpinnerModule
  ],
  declarations: [
    UserFormComponent
  ],
  exports: [
    UserFormComponent
  ]
})
export class UserFormModule { }
