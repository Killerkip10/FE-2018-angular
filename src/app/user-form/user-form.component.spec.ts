import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import {CommonModule} from '@angular/common';
import {TranslateModule} from '@ngx-translate/core';
import {ReactiveFormsModule} from '@angular/forms';
import {HttpClientModule} from '@angular/common/http';

import {UserFormComponent} from './user-form.component';
import {HttpValidatorService} from './validators';
import {AppTranslate} from '../app.translate';
import {User} from '../../../server/models';
import {Options} from './user-form.component';

describe('UserFormComponent', () => {
  let component: UserFormComponent;
  let fixture: ComponentFixture<UserFormComponent>;
  const user: User = {
    id: '23456789',
    login: '1234',
    password: '4321',
    name: 'Vlad',
    age: '19',
    role: '1',
    dateOfNextNot: '2018-11-10T18:52:23.023Z',
    dateOfFirstLogin: '2018-11-10T18:52:23.023Z',
    dateOfBirth: '2018-11-10T18:52:23.023Z',
    information: 'may the force be with you'
  };
  const options: Options = {
    nameForm: 'MAIN',
    send: 'SEND',
    reset: true,
    login: false,
    dateOfFirstLogin: false
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        CommonModule,
        TranslateModule,
        ReactiveFormsModule,
        HttpClientModule,

        AppTranslate
      ],
      declarations: [
        UserFormComponent
      ],
      providers: [
        HttpValidatorService
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserFormComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.user = user;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.user = user;
    component.options = options;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    const error = {message: 'TEST-ERROR'};

    component.userForm.controls.login.errors.login = error;
    component.userForm.controls.password.errors.password = error;
    component.userForm.controls.name.errors.name = error;
    component.userForm.controls.age.errors.age = error;
    component.userForm.controls.dateOfBirth.errors.date = error;
    component.userForm.controls.dateOfFirstLogin.errors.date = error;
    component.userForm.controls.dateOfNextNot.errors.date = error;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
