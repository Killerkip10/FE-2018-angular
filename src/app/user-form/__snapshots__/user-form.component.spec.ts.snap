// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFormComponent snapshot 1`] = `
<app-user-form
  fb={[Function FormBuilder]}
  loginAsyncValidService={[Function LoginAsyncValidatorService]}
  nameAsyncValidService={[Function NameAsyncValidatorService]}
  options={[Function Object]}
  submit={[Function EventEmitter]}
  user={[Function Object]}
  userForm={[Function FormGroup]}
>
  <mat-card>
    <mat-toolbar>
      
    </mat-toolbar>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="login"
              matinput=""
              ng-reflect-name="login"
              placeholder="USER-PROFILE.FORM.LOGIN"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              ng-reflect-name="password"
              placeholder="USER-PROFILE.FORM.PASSWORD"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="name"
              matinput=""
              ng-reflect-name="name"
              placeholder="USER-PROFILE.FORM.NAME"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="age"
              matinput=""
              ng-reflect-name="age"
              placeholder="USER-PROFILE.FORM.AGE"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfBirth"
              matinput=""
              ng-reflect-name="dateOfBirth"
              placeholder="USER-PROFILE.FORM.DATE-OF-BIRTH"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfFirstLogin"
              matinput=""
              ng-reflect-name="dateOfFirstLogin"
              placeholder="USER-PROFILE.FORM.DATE-OF-FIRST-LOGIN"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfNextNot"
              matinput=""
              ng-reflect-name="dateOfNextNot"
              placeholder="USER-PROFILE.FORM.DATE-OF-NEXT-NOT"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <textarea
              class="ng-untouched ng-pristine ng-valid"
              formcontrolname="information"
              matinput=""
              ng-reflect-name="information"
              placeholder="USER-PROFILE.FORM.INFORMATION"
            >
                        
            </textarea>
          </mat-form-field>
        </div>
        <div
          class="legend"
        >
          <div>
            USER-PROFILE.FORM.DATE-FORMATS: 
          </div>
          <div>
            YYYY/MM/DD | DD MMMM YYYY | DD-MMM-YY
          </div>
        </div>
        <button
          color="primary"
          disabled=""
          mat-flat-button=""
          type="submit"
        >
          
        </button>
      </form>
    </mat-card-content>
  </mat-card>
</app-user-form>
`;

exports[`UserFormComponent snapshot 2`] = `
<app-user-form
  fb={[Function FormBuilder]}
  loginAsyncValidService={[Function LoginAsyncValidatorService]}
  nameAsyncValidService={[Function NameAsyncValidatorService]}
  options={[Function Object]}
  submit={[Function EventEmitter]}
  user={[Function Object]}
  userForm={[Function FormGroup]}
>
  <mat-card>
    <mat-toolbar>
      
    </mat-toolbar>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="login"
              matinput=""
              ng-reflect-name="login"
              placeholder="USER-PROFILE.FORM.LOGIN"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              ng-reflect-name="password"
              placeholder="USER-PROFILE.FORM.PASSWORD"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="name"
              matinput=""
              ng-reflect-name="name"
              placeholder="USER-PROFILE.FORM.NAME"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="age"
              matinput=""
              ng-reflect-name="age"
              placeholder="USER-PROFILE.FORM.AGE"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfBirth"
              matinput=""
              ng-reflect-name="dateOfBirth"
              placeholder="USER-PROFILE.FORM.DATE-OF-BIRTH"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfFirstLogin"
              matinput=""
              ng-reflect-name="dateOfFirstLogin"
              placeholder="USER-PROFILE.FORM.DATE-OF-FIRST-LOGIN"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfNextNot"
              matinput=""
              ng-reflect-name="dateOfNextNot"
              placeholder="USER-PROFILE.FORM.DATE-OF-NEXT-NOT"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <textarea
              class="ng-untouched ng-pristine ng-valid"
              formcontrolname="information"
              matinput=""
              ng-reflect-name="information"
              placeholder="USER-PROFILE.FORM.INFORMATION"
            >
                        
            </textarea>
          </mat-form-field>
        </div>
        <div
          class="legend"
        >
          <div>
            USER-PROFILE.FORM.DATE-FORMATS: 
          </div>
          <div>
            YYYY/MM/DD | DD MMMM YYYY | DD-MMM-YY
          </div>
        </div>
        <button
          color="primary"
          disabled=""
          mat-flat-button=""
          type="submit"
        >
          
        </button>
      </form>
    </mat-card-content>
  </mat-card>
</app-user-form>
`;

exports[`UserFormComponent snapshot 3`] = `
<app-user-form
  fb={[Function FormBuilder]}
  loginAsyncValidService={[Function LoginAsyncValidatorService]}
  nameAsyncValidService={[Function NameAsyncValidatorService]}
  options={[Function Object]}
  submit={[Function EventEmitter]}
  user={[Function Object]}
  userForm={[Function FormGroup]}
>
  <mat-card>
    <mat-toolbar>
      MAIN
    </mat-toolbar>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="login"
              matinput=""
              ng-reflect-name="login"
              placeholder="USER-PROFILE.FORM.LOGIN"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              ng-reflect-name="password"
              placeholder="USER-PROFILE.FORM.PASSWORD"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="name"
              matinput=""
              ng-reflect-name="name"
              placeholder="USER-PROFILE.FORM.NAME"
              type="text"
            />
            
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="age"
              matinput=""
              ng-reflect-name="age"
              placeholder="USER-PROFILE.FORM.AGE"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfBirth"
              matinput=""
              ng-reflect-name="dateOfBirth"
              placeholder="USER-PROFILE.FORM.DATE-OF-BIRTH"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfFirstLogin"
              matinput=""
              ng-reflect-name="dateOfFirstLogin"
              placeholder="USER-PROFILE.FORM.DATE-OF-FIRST-LOGIN"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfNextNot"
              matinput=""
              ng-reflect-name="dateOfNextNot"
              placeholder="USER-PROFILE.FORM.DATE-OF-NEXT-NOT"
              type="text"
            />
          </mat-form-field>
          
        </div>
        <div>
          <mat-form-field>
            <textarea
              class="ng-untouched ng-pristine ng-valid"
              formcontrolname="information"
              matinput=""
              ng-reflect-name="information"
              placeholder="USER-PROFILE.FORM.INFORMATION"
            >
                        
            </textarea>
          </mat-form-field>
        </div>
        <div
          class="legend"
        >
          <div>
            USER-PROFILE.FORM.DATE-FORMATS: 
          </div>
          <div>
            YYYY/MM/DD | DD MMMM YYYY | DD-MMM-YY
          </div>
        </div>
        <button
          color="primary"
          disabled=""
          mat-flat-button=""
          type="submit"
        >
          SEND
        </button>
      </form>
    </mat-card-content>
  </mat-card>
</app-user-form>
`;

exports[`UserFormComponent snapshot 4`] = `
<app-user-form
  fb={[Function FormBuilder]}
  loginAsyncValidService={[Function LoginAsyncValidatorService]}
  nameAsyncValidService={[Function NameAsyncValidatorService]}
  options={[Function Object]}
  submit={[Function EventEmitter]}
  user={[Function Object]}
  userForm={[Function FormGroup]}
>
  <mat-card>
    <mat-toolbar>
      
    </mat-toolbar>
    <mat-card-content>
      <form
        class="ng-untouched ng-pristine ng-invalid"
        ng-reflect-form="[object Object]"
        novalidate=""
      >
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="login"
              matinput=""
              ng-reflect-name="login"
              placeholder="USER-PROFILE.FORM.LOGIN"
              type="text"
            />
            
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="password"
              matinput=""
              ng-reflect-name="password"
              placeholder="USER-PROFILE.FORM.PASSWORD"
              type="text"
            />
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="name"
              matinput=""
              ng-reflect-name="name"
              placeholder="USER-PROFILE.FORM.NAME"
              type="text"
            />
            
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="age"
              matinput=""
              ng-reflect-name="age"
              placeholder="USER-PROFILE.FORM.AGE"
              type="text"
            />
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfBirth"
              matinput=""
              ng-reflect-name="dateOfBirth"
              placeholder="USER-PROFILE.FORM.DATE-OF-BIRTH"
              type="text"
            />
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfFirstLogin"
              matinput=""
              ng-reflect-name="dateOfFirstLogin"
              placeholder="USER-PROFILE.FORM.DATE-OF-FIRST-LOGIN"
              type="text"
            />
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <input
              class="ng-untouched ng-pristine ng-invalid"
              formcontrolname="dateOfNextNot"
              matinput=""
              ng-reflect-name="dateOfNextNot"
              placeholder="USER-PROFILE.FORM.DATE-OF-NEXT-NOT"
              type="text"
            />
          </mat-form-field>
          
          <p>
            TEST-ERROR
          </p>
        </div>
        <div>
          <mat-form-field>
            <textarea
              class="ng-untouched ng-pristine ng-valid"
              formcontrolname="information"
              matinput=""
              ng-reflect-name="information"
              placeholder="USER-PROFILE.FORM.INFORMATION"
            >
                        
            </textarea>
          </mat-form-field>
        </div>
        <div
          class="legend"
        >
          <div>
            USER-PROFILE.FORM.DATE-FORMATS: 
          </div>
          <div>
            YYYY/MM/DD | DD MMMM YYYY | DD-MMM-YY
          </div>
        </div>
        <button
          color="primary"
          disabled=""
          mat-flat-button=""
          type="submit"
        >
          
        </button>
      </form>
    </mat-card-content>
  </mat-card>
</app-user-form>
`;
